jb/book_cache_aware_stats.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef jb_book_cache_aware_stats_hpp
+#define jb_book_cache_aware_stats_hpp
+
+#include <jb/itch5/order_book_def.hpp>
+#include <jb/config_object.hpp>
+#include <jb/event_rate_histogram.hpp>
+#include <jb/histogram.hpp>
+#include <jb/integer_range_binning.hpp>
+
+#include <iosfwd>
+
+namespace jb {
+
+typedef jb::itch5::tick_t tick_t;
+typedef jb::itch5::level_t level_t;
+
+/**
+ * Keep statistics about changes on the inside
+ * of a book, as well as tail movements
+ */
+class book_cache_aware_stats {
```

Comment:
I think we are doing this wrong if we need a new stats class every time a new book type is created.  Maybe we could create a generic "print_csv()" method that takes the cutting points as an input?

Something like:

    template<typename sample_type>
    class histogram_summary {
    public:
      explicit histogram_summary(vector<double> const& buckets);
      void print_csv_header(std::ostream const&, std::string const& suffix) const {
        for each bucket {
          os << "," << "p" + pretty_format_percentile(bucket) + suffix;
        }
     }
     void println_csv_header(...) {
       print_csv_header(...); os << "\n";
    }